### PR TITLE
2918-list-of-granted-consents

### PR DESCRIPTION
**Pull request by sahma19**

### CODE REVIEW

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+﻿{}
```
Comment (ckr123) [inline domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt:1]: Should this be 
```
{
    Result: []
}
```
?

File: domains/authorization/Authorization.API/API/Authorization/Controllers/ConsentController.cs [hunk 2]
```diff
@@ -46,7 +40,20 @@ public async Task<ActionResult> GrantConsent([FromServices] ILogger<ConsentContr
...
+    /// <summary>
+    /// Retrieves consents granted by the organization that a user is affiliated with. It will read the IdpUserId and OrgCvr claims from the users session token, use those to query the database, and return a list of consents.
+    /// </summary>
+    [HttpGet]
+    [Route("api/authorization/consents/")]
+    public async Task<ActionResult> GetConsent([FromServices] ILogger<ConsentController> logger)
+    {
+        var identity = new IdentityDescriptor(HttpContext);
+
+        var result = await mediator.Send(new GetUserOrganizationConsentsQuery(identity.Sub.ToString(), identity.OrgCvr!));
```
Comment (martinhenningjensen) [inline domains/authorization/Authorization.API/API/Authorization/Controllers/ConsentController.cs:56]: Why cvr and not organization id?

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+        await dbContext.Affiliations.AddRangeAsync([affiliation1, affiliation2]);
+        await dbContext.Consents.AddRangeAsync([consent1, consent2]);
+
+        await dbContext.SaveChangesAsync();
+
+        var userIdString = user.IdpUserId.Value.ToString();
+
+        var userClient = _integrationTestFixture.WebAppFactory.CreateApi(sub: userIdString);
+        var response = await userClient.GetUserOrganizationConsents();
```
Comment (martinhenningjensen) [inline domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs:103]: This and the check for 200 OK can be combined into a single GetFromJsonAsync method call.

**Review by martinhenningjensen — CHANGES_REQUESTED** [review]

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+        await Verify(deserializedResponse);
+    }
+
+    [Fact]
+    public async Task GivenUserAffiliatedWithMultipleOrganizations_WhenGettingConsent_ThenOnlyConsentFromCurrentOrganizationContextIncludedInResponse()
+    {
+        var user = Any.User();
+        var organization1 = Any.Organization();
+        var organization2 = Any.Organization(Tin.Create("87654321"));
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs:80]: The handler queries the database using the users orgCvr. By changing the clients orgCvr, i make sure the user is not affiliated with the organization.

File: domains/authorization/Authorization.API/API/Authorization/Controllers/ConsentController.cs (hunk 2 above)
Comment (sahma19) [reply domains/authorization/Authorization.API/API/Authorization/Controllers/ConsentController.cs:56]: Does it have to be orgId?

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        var affiliation2 = Affiliation.Create(user, organization2);
+
+        await dbContext.Affiliations.AddRangeAsync([affiliation1, affiliation2]);
+        await dbContext.Consents.AddRangeAsync([consent1, consent2]);
+
+        await dbContext.SaveChangesAsync();
+
+        var userIdString = user.IdpUserId.Value.ToString();
+
+        var userClient = _integrationTestFixture.WebAppFactory.CreateApi(sub: userIdString);
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs:87]: No. by default both users will end up with same orgCvr, which is why it works

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    {
+        await SeedData();
+
+        var response = await _api.GetUserOrganizationConsents();
+
+        response.Should().Be200Ok();
+
+        var deserializedResponse = JsonConvert.DeserializeObject<GetUserOrganizationConsentsQueryResult>(await response.Content.ReadAsStringAsync());
+
+        await Verify(deserializedResponse);
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs]: done

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        var response = await userClient.GetUserOrganizationConsents();
+
+        response.Should().Be200Ok();
+
+        var result = await response.Content.ReadFromJsonAsync<GetUserOrganizationConsentsQueryResult>();
+        result!.Result.Should().NotBeEmpty();
+    }
+
+    [Fact]
+    public async Task GivenUser_WhenGettingConsent_ThenVerifyResponseSnapshot()
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs]: done

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenUser_WhenGettingConsent_ThenVerifyResponseSnapshot.verified.txt
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenUser_WhenGettingConsent_ThenVerifyResponseSnapshot.verified.txt:1]: done

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt:1]: done

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt (hunk 1 above)
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt]: Do our other apis' return a result array?

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (martinhenningjensen) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs:65]: Isn't affiliation in the data given by an affiliation entity? The user could still be affiliated to a company with a different CVR?

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs [hunk 8]
```diff
@@ -57,12 +57,28 @@ public async Task GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBod
...
+    [Fact]
+    public async Task GivenNoConsentsExist_WhenUserQueriesForConsents_VerifyEmptyBodyFormat()
+    {
+        await SeedData();
+
+        var response = await _api.GetUserOrganizationConsents();
+
+        response.Should().Be200Ok();
+
+        var deserializedResponse = JsonConvert.DeserializeObject<GetUserOrganizationConsentsQueryResult>(await response.Content.ReadAsStringAsync());
```
Comment (TopSwagCode) [inline domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.cs:70]: var response = await _api.GetClientConsents();
        var result = await response.Content.ReadFromJsonAsync<ClientConsentsResponse>();

File: domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt (hunk 1 above)
Comment (sahma19) [reply domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/GetConsentTests.GivenNoConsentsExist_WhenUserQueriesForConsents_HttpOKEmptyBody.verified.txt]: Resolving comment with permission from @ckr123

**Review by martinhenningjensen — APPROVED** [review]

**Review by martinhenningjensen — APPROVED** [review]

**Pull request merged**